server/db/db.js

```diff
@@ -150,6 +152,32 @@ async function getClients(pgClient) {
   }
 }
 
+async function getLocations(pgClient) {
+  try {
+    const results = await helpers.runQuery(
+      'getLocations',
+      `
+      SELECT l.*
+      FROM locations AS l
+      LEFT JOIN clients AS c ON l.client_id = c.id
+      ORDER BY c.display_name, l.display_name
+      `,
+      [],
+      pool,
+      pgClient,
+    )
+
+    if (results === undefined) {
+      return null
+    }
+
+    const allClients = await getClients(pgClient)
+    return results.rows.map(r => createLocationFromRow(r, allClients))
+  } catch (err) {
+    helpers.log(err.toString())
```

Comment:
I would suggest logging the entire error using helper.log(err) instead of its string representation. This would provide us with more detailed error information

Reply:
This is a good thing to experiment with in the future. Right now, this is basically boilerplate code that is used for all the database functions in this file and in the Button's `db.js` as well. So I don't want to change it. But I could see a project in the future to examine exactly what we want to log and then changing it for all the functions if that's what you decide to do.

Some things to consider:
1. I think that `helpers.log` takes a string value. So if you just send `helpers.log(err)` then it will just output `[Object object]` which is less helpful than `err.toString()`
2. Sometimes the full error contains information that would be a security risk if it were in the logs. For example, the DB password. If this error object contains the DB password in the logs, and that was sent to Sentry and stored on the server logs in Digital Ocean (or AWS) then that means it now exists in at least 3 places. Therefore, we are three times as likely to get hacked and our password compromised. 

So I agree that more information can be useful to us. This is outside the scope of this PR for the moment. If this is something that you'd like to look into, it would be cool if you make a ClickUp Task in the Backlog of the Devices Kanban, and it can be prioritized along with the other Tasks at our next Prioritization Meeting. Thanks!